### PR TITLE
Use icotool to generate .ico files instead of png2ico

### PR DESCRIPTION
**Pull request by dschmidt**

### CODE REVIEW

File: kde/frameworks/extra-cmake-modules/extra-cmake-modules.py [hunk 1]
```diff
@@ -5,9 +5,13 @@ class subinfo(info.infoclass):
     def setTargets(self):
         self.versionInfo.setDefaultValues()
 
+        for ver in self.versionInfo.tarballs() + self.versionInfo.branches() + self.versionInfo.tags():
+            ecmVer = CraftVersion(ver)
```
Comment (TheOneRing) [inline kde/frameworks/extra-cmake-modules/extra-cmake-modules.py]: not used

File: kde/frameworks/extra-cmake-modules/icotool.diff [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                     WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+                 )
+-            endif()
++        endfunction()
++
++        if (IcoTool_FOUND)
++            set(icotool_args "-c -o \"${_outfilename}.ico\"")
++
++            foreach(size 16 32 48)
++                foreach(icon ${icons_at_${size}px})
```
Comment (TheOneRing) [inline kde/frameworks/extra-cmake-modules/icotool.diff:109]: what about 48 < x < 256

File: kde/frameworks/extra-cmake-modules/extra-cmake-modules.py (hunk 1 above)
Comment (dschmidt) [reply kde/frameworks/extra-cmake-modules/extra-cmake-modules.py]: ah oops. Should I use [ver] and get rid of this var or use this var?

File: kde/frameworks/extra-cmake-modules/icotool.diff (hunk 2 above)
Comment (dschmidt) [reply kde/frameworks/extra-cmake-modules/icotool.diff:109]: They are not part of the "classic full set".

https://msdn.microsoft.com/en-us/library/windows/desktop/dn742485%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396

```
Application icons and Control Panel items: The full set includes 16x16, 32x32, 48x48, and 256x256 (code scales between 32 and 256). The .ico file format is required. For Classic Mode, the full set is 16x16, 24x24, 32x32, 48x48 and 64x64.
```

As I understand it, these are the only sizes that are supported for *application* icons. For other usages this macro is not made and would need to be adjusted anyway.

File: kde/frameworks/extra-cmake-modules/extra-cmake-modules.py (hunk 1 above)
Comment (TheOneRing) [reply kde/frameworks/extra-cmake-modules/extra-cmake-modules.py]: yes the craft version in qtbase is only used for the version check

File: kde/frameworks/extra-cmake-modules/icotool.diff (hunk 2 above)
Comment (TheOneRing) [reply kde/frameworks/extra-cmake-modules/icotool.diff:109]: hm sounds good

**Pull request merged**